internal/controller/volumereplicationgroup_controller.go

```diff
@@ -1198,7 +1204,7 @@ func (v *VRGInstance) processAsPrimary() ctrl.Result {
 			v.log.Info("Kube objects restore failed", "error", err)
 			v.errorConditionLogAndSet(err, "Failed to restore kube objects", setVRGKubeObjectsErrorCondition)
 
-			return v.updateVRGStatus(v.result)
+			return v.updateVRGConditionsAndStatus(v.result)
 		}
 
 		v.log.Info("Kube objects restored")
```

Comment:
Follow the `shouldRestoreClusterData` pattern and update status here, before progressing with further processing (which includes a new backup and VRG upload conditionally).

As e2e reports an error where restore of k8s resources is triggered twice, at this point returning to update status to ensure no further processing is done would be better as suggested by @raghavendra-talur